mix/mix_protocol.nim

```diff
@@ -269,14 +312,18 @@ proc buildSurbs(
     response.add(surb)
 
   if surbSK.len != 0:
-    mixProto.idToSKey[id] = surbSK
+    mixProto.connCreds[id] = ConnCreds(surbSKSeq: surbSK, incoming: incoming)
 
   return ok(response)
 
 proc prepareMsgWithSurbs(
-    mixProto: MixProtocol, msg: seq[byte], numSurbs: uint8 = 0, skipPeer: PeerId
+    mixProto: MixProtocol,
+    incoming: AsyncQueue[seq[byte]],
```

Comment:
Similar to comment in `buildSurbs`:
- if `incoming` is always a fresh queue, might be simpler to create  internally here too?
- `skipPeer` seems a bit ambiguous.